### PR TITLE
Use Ruff for linting

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,8 +22,13 @@ repos:
         files: \.(md|rst|yml|yaml)
         args: [--prose-wrap=preserve]
 
-  - repo: https://github.com/asottile/pyupgrade
```
Comment (jarrodmillman) [inline .pre-commit-config.yaml:25]: Why remove pyupgrade?

**Comment by jarrodmillman** [comment]
Would this replace pep8speaks?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (stefanv) [reply .pre-commit-config.yaml:25]: The pyupgrade rules are built into Ruff.

**Comment by stefanv** [comment]
> Would this replace pep8speaks?

This could replace pep8speaks; the difference is mainly where it lets you know about issues.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jarrodmillman) [reply .pre-commit-config.yaml:25]: It looks like it only implements some of pyupgrade, but I couldn't find a list of what it doesn't implement.

For example, how do you specify what the minimum version of Python you support is? I couldn't figure it out.

**Comment by jarrodmillman** [comment]
I always found pep8speaks more annoying than helpful. I will be happy to see if gone.

**Comment by hmaarrfk** [comment]
ruff is great when running locally. We have "run linting" script. When using flake8, it was painful, now with ruff, due to the increased speed, we are more comfortable running it all the time.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (stefanv) [reply .pre-commit-config.yaml:25]: Take a look at the pyproject.toml

@charliemarsh may be able to comment on which rules are in and out of scope.

**Comment by jarrodmillman** [comment]
The `Test / linux-cp3.9-install-from-sdist (pull_request)` is incorrectly marked as passing. This PR should fix the test: PR #6731.

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -1,6 +1,4 @@
-# .git-blame-ignore-revs
```
Comment (jarrodmillman) [inline .git-blame-ignore-revs:1]: Don't we want to keep the old hashes and we will need to update the new list after the PR is merged into main.

**Review by jarrodmillman — APPROVED** [review]
LGTM

File: .git-blame-ignore-revs (hunk 2 above)
Comment (stefanv) [reply .git-blame-ignore-revs:1]: Fixed this, and added the new commit hash.

**Pull request merged**